IMS.BL/Application/Program.cs

```diff
@@ -88,12 +94,9 @@ public static void Main(string[] args)
             }
             catch (Exception e)
             {
-                if (e.GetType() == typeof(NullReferenceException))
-                {
-                    Console.WriteLine("There are no products with the provided data!");
-                }
-                
-                Console.WriteLine("The data you entered is not valid, please try again and provide a valid data!");
+                Console.WriteLine(e.GetType() == typeof(NullReferenceException)
+                    ? "There are no products with the provided data!"
+                    : "The data you entered is not valid, please try again and provide a valid data!");
                 Environment.Exit(1);
```

Comment:
if you ends the program each time there is a problem then your program is useless and not fault tolerant, its not a good user experience to exit the program if i asked of list of products and there wastnt any of then so you kick me out of the program, fix the fault tolerant issues in your program and make more custom exceptions to deliver more meaningful error messages for the user